README.md

```diff
@@ -302,19 +302,28 @@ The above walkthrough follows the actual implementation process used to produce
+The previous section has covered manually running an operator. In the next sections, we will explore using the [Operator Lifecycle Manager][operator_lifecycle_manager] (OLM) which is what enables a more robust deployment model for operators being run in production environments.
+
+OLM helps you to install, update, and generally manage the lifecycle of all of the operators (and their associated services) on a Kubernetes cluster. It runs as an Kubernetes extension and lets you use `kubectl` for all the lifecycle management functions without any additional tools.
+
 
-The Operator Lifecycle Manager helps you to install, update, and generally manage the lifecycle of all of the operators (and their associated services) on a Kubernetes cluster. It runs as an Kubernetes extension and lets you use `kubectl` for all the lifecycle management functions without any additional tools.
 
 ### Generate an operator manifest
 
-The first step to leveraging the Operator Lifecycle Manager is to create a manifest. An operator manifest describes how to display, create and manage the application, in this case memcached, as a whole. It is required for the Operator Lifecycle Manager to function.
+The first step to leveraging OLM is to create a [Cluster Service Version][csv_design_doc] (CSV) manifest. An operator manifest describes how to display, create and manage the application, in this case memcached, as a whole. It is required for OLM to function.
+
+The Operator SDK CLI can generate CSV manifests via the following command:
+```console
+$ operator-sdk olm-catalog gen-csv --csv-version 0.0.1
+```
+For more details see the SDK's [CSV generation doc][csv_generation_doc].
+
 
-For the purpose of this guide, we will continue with this [predefined manifest][manifest_v1] file for the next steps. If you’d like, you can alter the image field within this manifest to reflect the image you built in previous steps, but it is unnecessary. In the future, the Operator SDK CLI will generate an operator manifest for you, a feature that is planned for the next release of the Operator SDK.
+For the purpose of this guide, we will continue with this [predefined manifest][manifest_v1] file for the next steps. If you’d like, you can alter the image field within this manifest to reflect the image you built in previous steps, but it is unnecessary.
 
 ### Deploy the Operator
 
-Deploying an operator is as simple as applying the operator’s manifest to the desired namespace in the cluster.
+Deploying an operator is as simple as applying the operator’s CSV manifest to the desired namespace in the cluster.
 
 ```sh
 $ curl -Lo memcachedoperator.0.0.1.csv.yaml https://raw.githubusercontent.com/operator-framework/getting-started/master/memcachedoperator.0.0.1.csv.yaml
```

Comment:
Creating a CSV in-cluster using the `kubectl apply/create` method requires an `OperatorGroup` in that CSV's namespace. Additionally, `ServiceAccount`s, `Role`s, and `RoleBinding`'s should be created in that namespace.

Also instead of using the `getting-started` repo's memcached CSV, I recommend using the samples repo CSV:
```console
$ curl -Lo memcachedoperator.csv.yaml https://raw.githubusercontent.com/operator-framework/operator-sdk-samples/memcached-operator/deploy/olm-catalog/memcached-operator/0.6.0/memcached-operator.v0.6.0.clusterserviceversion.yaml
```

I'll update the samples repo CSV to have a different namespace than `placeholder` so the CSV creation flow makes more sense to users.

Reply:
@estroz Yeah I wanted to replace the CSV examples in this repo with the CSVs from samples operator.

One reason I can't do that right now is because our example CSV doesn't have an operator image right now so the walkthrough won't work as is.
https://github.com/operator-framework/operator-sdk-samples/blob/master/memcached-operator/deploy/olm-catalog/memcached-operator/0.5.0/memcached-operator.v0.5.0.clusterserviceversion.yaml#L47

I'll have to add a step to mention to change the image field to their own which is one more  step for the user. Or I was thinking we could maintain an image for our sample memcached-operator at `quay.io/operator-framework/sample-memcached-operator` and just have our sample CSV point to that so the walkthrough works without requiring users to build and push their own images.

So I'll have to do a follow up PR for that once we maintain our own image for that.


> Additionally, ServiceAccounts, Roles, and RoleBinding's should be created in that namespace

This is for the operator right? We do that right after creating the CSV in lines 337-348
https://github.com/operator-framework/getting-started/pull/36/files#diff-04c6e90faac2675aa89e2176d2eec7d8R338
Is that what you meant?

Reply:
I think maintaining a sample image would make sense, or use the one pushed by the test framework CSV (`quay.io/coreos/operator-sdk-dev:test-framework-operator`).

Those 4 lines should be
```console
$ kubectl create -f ... --namespace default
```

Reply:
@hasbro17 I think it might make sense to advertise operator-sdk's generate csv command in this case. If we want people to buy into the operator-framework, CSVs and OLM, then we should push these tools we've created as we expect them to be used in an actual workflow.
Providing a sample CSV is good for reference is manual steps are still required, but it also sets up the expectation that someone from the framework team will always be able to help with creating an operator's CSV, which (although our current approach) isn't sustainable.
Are there any objections to adding steps for using the CSV generation command into the getting started? I can submit this as a follow-on PR.

Reply:
@SamiSousa I've mentioned the CSV generation command above in line 317 but you can probably expand on that in a follow up PR.
I agree that we should make it part of the workflow to generate the CSV and use that for the remainder of the guide.

There's one more step that's still needed after generating your own CSV and that's to specify the operator image.
As Eric and I've discussed above we're thinking of asking users to either use an image that we've maintained for them (e.g `quay.io/operator-framework/sample-memcached-operator` built off the samples memcached-operator).
Or we can ask users to push their image first (which they've done in the SDK section) and then specify that in the CSV. WDYT?

@estroz The 4 lines are already in this PR. I'm not using the `--namespace=default` flag because users can have the default ns set in their kubeconfig to something else:
```console
$ kubectl create -f deploy/crds/cache_v1alpha1_memcached_crd.yaml
$ kubectl create -f deploy/service_account.yaml
$ kubectl create -f deploy/role.yaml
$ kubectl create -f deploy/role_binding.yaml
```
The only thing missing is creating the operator group since the SDK doesn't generate the manifest for that. That should just be the following I think:
```YAML
apiVersion: operators.coreos.com/v1alpha2
kind: OperatorGroup
metadata:
  name: memcached-operator-group
  namespace: default
  spec:
    targetNamespaces:
    - default
```
I'll add a step for that.